### PR TITLE
Fixes eth call error propagation

### PR DESCRIPTION
**Pull request by otherview**
### Why is this change needed?

- Errors were not being correctly propagated from the EVM to the user

### What changes were made as part of this PR:

- Removed error wrapping

### :rotating_light: Definition of done :rotating_light:
- [ ] Unit tests added to cover new or changed functionality 
- [ ] Docs pages updated to cover new or changed functionality
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated

### CODE REVIEW

File: go/enclave/evm/evm_facade.go [hunk 1]
```diff
@@ -5,14 +5,8 @@ import (
 	"fmt"
 	"math"
 
-	gethlog "github.com/ethereum/go-ethereum/log"
```
Comment (otherview) [inline go/enclave/evm/evm_facade.go:8]: Imports reordered

File: go/enclave/rpc_server.go [hunk 2]
```diff
@@ -7,23 +7,20 @@ import (
 	"fmt"
 	"net"
 
-	"github.com/obscuronet/go-obscuro/go/common/log"
```
Comment (otherview) [inline go/enclave/rpc_server.go:10]: Imports reordered but not changed

File: go/enclave/evm/evm_facade.go [hunk 3]
```diff
@@ -94,18 +93,22 @@ func ExecuteOffChainCall(msg *types.Message, s *state.StateDB, header *common.He
...
+	// 1 - vmError / stateDB err check
+	// 2 - evm.Cancelled() TODO
+	// 3 - error check the ApplyMessage
 	// Read the error stored in the database.
 	err = s.Error()
 	if err != nil {
 		return nil, newErrorWithReasonAndCode(err)
 	}
 
+	if err != nil {
```
Comment (tudor-malene) [inline go/enclave/evm/evm_facade.go:106]: this block should be after the revert, and the "err" should prob be called something else

**Review by tudor-malene — CHANGES_REQUESTED** [review]

File: go/enclave/evm/evm_facade.go (hunk 3 above)
Comment (tudor-malene) [reply go/enclave/evm/evm_facade.go:106]: still, this check needs to go last I'd say. to be consistent with geth

Comment (otherview) [reply go/enclave/evm/evm_facade.go:106]: Changed the name of the dbErr 👍 

I thought it was high-to-low error checking. VM, then Execution, then Result err checking. But I might have misunderstood that.

Comment (otherview) [reply go/enclave/evm/evm_facade.go:111]: I don't think geth checks for Result reverse in this method, it's only when it's returning the payload.
https://github.com/ethereum/go-ethereum/blob/5a02b2d6d0a8d5c8ebf170505c681f7b19df7ed9/internal/ethapi/api.go#L1049

But logically speaking is the same thing, so I think it's fine to have this order of checks.

**Review by tudor-malene — APPROVED** [review]
lgtm

**Pull request merged**